front/app/routes/_main/map/$placeId_.$heistId.edit/route.tsx

```diff
@@ -0,0 +1,353 @@
+    });
+  }
+}
+
+export type Loader = typeof loader;
+
+export async function action({ request, context, params }: ActionFunctionArgs) {
+  const user = denyAccessUnlessGranted(context.user, ROLES.CONTRACTOR);
+  const isAdmin = hasRoles(context.user, ROLES.ADMIN);
+
+  if (!params?.heistId) {
+    throw redirect(`/map/${params.placeId}`);
+  }
+
+  const isPublic = await heistIsPublic(context.client, params.heistId);
+
+  if (isPublic) {
+    return redirect(`/map/${params.placeId}`);
+  }
+
+  const isMadeBy = await heistIsMadeBy(context.client, {
+    id: params.heistId,
+    userId: user.id,
+  });
+
+  if (!isMadeBy && !isAdmin) {
+    throw redirect(`/map/${params.placeId}`);
+  }
+
+  const t = await i18next.getFixedT(request, ['validators', 'flash']);
+  const { errors, data } = await getValidatedFormData<UpdateHeistFormData>(
+    request,
+    updateHeistResolver,
+  );
+
+  if (errors) {
+    return json({ errors }, { status: 400 });
+  }
+
+  let errorMessage: string | null = null;
+  const session = await getSession(request.headers.get('Cookie'));
+
+  try {
+    await updateHeist(context.client, {
+      ...data,
+      id: params?.heistId,
+      minimumPayout: +data.minimumPayout,
+      maximumPayout: +data.maximumPayout,
+      minimumRequiredRating: +(data?.minimumRequiredRating ?? 0),
+      startAt: dayjs(`${data.startAtDate} ${data.startAtTime}`).toISOString(),
+      shouldEndAt: dayjs(`${data.shouldEndAtDate} ${data.shouldEndAtTime}`)
+        .utc(false)
+        .toISOString(),
+      difficulty: data.difficulty.value,
+      preferedTactic: data.preferedTactic.value,
+      visibility: data.visibility.value,
+      allowedEmployees: data.allowedEmployees.map((allowedEmployee) => allowedEmployee.value),
+      forbiddenUsers: data.forbiddenUsers?.map((user) => user.value),
+      forbiddenAssets: data.forbiddenAssets?.map((asset) => asset.value),
+    });
+
+    session.flash(FLASH_MESSAGE_KEY, {
+      content: t('heist.updated_successfully', { ns: 'flash' }),
+      type: 'success',
+    } as FlashMessage);
+
+    return redirect(`/map/${params.placeId}`, {
+      headers: {
+        'Set-Cookie': await commitSession(session),
+      },
+    });
+  } catch (error) {
+    if (error instanceof ClientError && hasErrorStatusCodes(error, [422, 401])) {
+      errorMessage = getMessageForErrorStatusCodes(error, [422, 401]);
+    } else {
+      throw error;
+    }
+  }
+
+  if (errorMessage) {
+    session.flash(FLASH_MESSAGE_KEY, {
+      content: errorMessage,
+      type: 'error',
+    } as FlashMessage);
+  }
+
+  return json(
+    { errors: {} },
+    { status: 401, headers: { 'Set-Cookie': await commitSession(session) } },
+  );
+}
+
+type Option = { label: string; value: string };
+
+export default function Edit() {
+  const { t } = useTranslation();
+  const { placeId, heist, employees, assets, users, user } = useLoaderData<Loader>();
+
+  const usersFormatted: Option[] = users.edges.reduce((acc, curr) => {
```

Comment:
```suggestion
  const usersFormatted: Option[] = users.edges.reduce<Option>((acc, curr) => {
```

Je crois qu'on peut faire ça